firmware-support/bittide-sys/src/time.rs

```diff
@@ -0,0 +1,353 @@
+// SPDX-FileCopyrightText: 2023 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+/*! Time structures.
+
+The `time` module contains structures used to get some sense of time.
+While clocks are dynamic in a Bittide System, it is expected that the deviation from
+wall clock time is no worse than any on-board static clock source and the effect is
+insignificant for timing purposes.
+
+ - [`Instant`] is used to represent absolute time.
+ - [`Duration`] is used to represent relative time.
```

Comment:
```suggestion
 - [`Instant`] is used to represent absolute time.
 - [`Duration`] is used to represent relative time.
```

Perhaps go with Python terminology: `TimeStamp` and `TimeDelta`. Or, if this it is common in rust to scope it to the module: `Stamp` and `Delta`?